File: Data Structures & Algorithms/house-robber/submission-3.js

class Solution {
			rob(nums) {
		if (nums.length === 0) return 0;
		if (nums.length === 1) return nums[0];
		let table = new Array(nums.length).fill(0);
		table[0] = nums[0];
		table[1] = Math.max(nums[0], nums[1])
		for (let i = 2; i < nums.length; i++) {
			table[i] += Math.max(nums[i] + table[i - 2], table[i - 1]);
		}
		return table[nums.length - 1];



	}




	

}

File: Data Structures & Algorithms/maximum-subarray/submission-3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total = 0
        cur_mx= nums[0]
        for num in nums:
            if total<0:
                total = 0
            total+=num
            cur_mx = max(cur_mx , total)
        return cur_mx

